lib/pacemaker/pcmk_sched_colocation.c

```diff
@@ -195,23 +199,34 @@ pcmk__add_this_with(GList **list, const pcmk__colocation_t *colocation)
  *
  * \param[in,out] list      List of constraints to add \p addition to
  * \param[in]     addition  List of colocation constraints to add to \p list
+ * \param[in]     rsc       Resource whose colocations we're getting (for
+ *                          logging only)
  *
  * \note The lists must be pre-sorted by cmp_primary_priority().
  */
 void
-pcmk__add_this_with_list(GList **list, GList *addition)
+pcmk__add_this_with_list(GList **list, GList *addition,
+                         const pe_resource_t *rsc)
 {
-    CRM_CHECK((list != NULL), return);
-
-    if (*list == NULL) { // Trivial case for efficiency
-        crm_trace("Copying %u 'this with' colocations to new list",
-                  g_list_length(addition));
-        *list = g_list_copy(addition);
-    } else {
-        while (addition != NULL) {
-            pcmk__add_this_with(list, addition->data);
-            addition = addition->next;
+    CRM_ASSERT((list != NULL) && (rsc != NULL));
+
+    pcmk__if_tracing(
+        {}, // Always add each colocation individually if tracing
+        {
+            if (*list == NULL) {
+                // Trivial case for efficiency if not tracing
+                pe_rsc_trace(rsc,
+                             "Copying %u 'this with' colocations to new list "
+                             "for %s",
+                             g_list_length(addition), rsc->id);
```

Comment:
We know we're not tracing here ... but someone can use PCMK_trace_formats so this is still theoretically useful

Reply:
I'll update the comment real quick. I meant it in contrast to the "if tracing" above -- so in the sense of "not tracing, so trivial case for efficiency"

Reply:
OH, the `pe_rsc_trace()`... crap

Reply:
We could either drop it (preferable) or make it debug (seems inappropriate). It seems silly to keep it on the off-chance someone is using a matching `PCMK_trace_formats`, honestly.

Reply:
Just drop it